server/runtime/src/main/java/org/infinispan/server/configuration/security/KeyStoreConfiguration.java

```diff
@@ -90,8 +92,8 @@ private KeyStore buildKeyStore(Properties properties) throws GeneralSecurityExce
             properties.getProperty(attributes.attribute(RELATIVE_TO).get()));
       String generateSelfSignedHost = attributes.attribute(GENERATE_SELF_SIGNED_CERTIFICATE_HOST).get();
       String provider = attributes.attribute(PROVIDER).get();
-      char[] keyStorePassword = attributes.attribute(KEYSTORE_PASSWORD).get().get();
-      char[] keyPassword = attributes.attribute(KEY_PASSWORD).isNull() ? null : attributes.attribute(KEY_PASSWORD).get().get();
+      char[] keyStorePassword = resolvePassword(attributes.attribute(KEYSTORE_PASSWORD));
+      char[] keyPassword = resolvePassword(attributes.attribute(KEY_PASSWORD));
       String keyAlias = attributes.attribute(ALIAS).get();
       if (!new File(keyStoreFileName).exists() && generateSelfSignedHost != null) {
          KeyStoreUtils.generateSelfSignedCertificate(keyStoreFileName, provider, keyStorePassword,
```

Comment:
can you double check `generateSelfSignedCertificate()` the `provider` parameter is ignored inside the method.

Reply:
you mean `KeyStoreUtils.generateSelfSignedCertificate()` ?

Reply:
yes.